src/ItemSorter.js

```diff
@@ -0,0 +1,55 @@
+export default class ItemSorter {
+  constructor(itemList, itemRepository) {
+    this.itemBeingDragged = null;
+    this.itemList = itemList;
+    this.itemRepository = itemRepository;
+  }
+
+  itemDragStart(e) {
+    if (!e.target.classList.contains('item')) {
+      return;
+    }
+
+    this.itemBeingDragged = e.target;
+    this.itemBeingDragged.classList.add('dragging');
+  }
+
+  itemDragOver(e) {
+    e.preventDefault();
+
+    if (!this.itemBeingDragged) {
+      return;
+    }
+
+    if (!e.target.classList.contains('item')) {
+      return;
+    }
+
+    const itemBeingDraggedOver = e.target;
+
+    if (this.itemBeingDragged.id === itemBeingDraggedOver.id) {
+      return;
+    }
+
+    const mouseVerticalPosition = e.clientY;
+    const itemBeingDraggedOverRectangle = itemBeingDraggedOver.getBoundingClientRect();
+    const itemBeingDraggedOverVerticalCenter = itemBeingDraggedOverRectangle.top
+                                            + (itemBeingDraggedOverRectangle.height / 2);
+
+    if (mouseVerticalPosition <= itemBeingDraggedOverVerticalCenter) {
+      this.itemBeingDragged.before(itemBeingDraggedOver);
+    } else {
+      this.itemBeingDragged.after(itemBeingDraggedOver);
+    }
+  }
```

Comment:
Good job here....

- [ ] your draggable functions is not working properly, I can't drag item from one point to the other, please fix this